rs/types/types/src/lib.rs

```diff
@@ -584,25 +585,65 @@ pub trait CountBytes {
 /// Allow an object to report its own byte size on disk and in memory. Not
 /// necessarily exact.
 pub trait MemoryDiskBytes {
-    fn memory_bytes(&self) -> usize;
-    fn disk_bytes(&self) -> usize;
-}
+    /// Returns the number of bytes that this object occupies in heap memory.
+    fn heap_bytes(&self) -> usize {
+        // No heap allocations by default.
+        0
+    }
 
-impl MemoryDiskBytes for Time {
+    /// Returns the total number of bytes that this object occupies
+    /// in stack and heap memory.
     fn memory_bytes(&self) -> usize {
```

Comment:
Ideally we'd make this implementation "frozen" so that people can't override it, but I guess that is not possible.

Reply:
WDYT?